app/uk/gov/hmrc/customerprofile/services/CustomerProfileService.scala

```diff
@@ -112,37 +114,38 @@ class CustomerProfileService @Inject() (
       } yield response
     }
 
-  private def mapStatusToExistingValue(statusReceived: Option[Status]): Option[StatusName] = statusReceived match {
-    case Some(Status.Pending)  => Some(StatusName.Pending)
-    case Some(Status.Bounced)  => Some(StatusName.Bounced)
-    case Some(Status.Verified) => Some(StatusName.Verified)
-    case None                  => None
-  }
+  def reOptInEnabledCheck(preferences: Preference): Preference =
+    if (reOptInEnabled)
+      preferences
+    else {
+      preferences.status.map(_.name) match {
+        case Some(statusName) =>
+          if (statusName.contains(ReOptIn))
+            preferences.copy(email = preferences.email.map(_.copy(status = Verified)),status = preferences.status.map(_.copy(name = Some(Verified))))
+          else preferences
+        case _ => preferences
+      }
+    }
 
-  private def makePreferencesForwardsCompatible(
+  private def makePreferencesBackwardsCompatible(
     preferencesReceived: Future[Option[Preference]]
   )(implicit ex:         ExecutionContext
   ): Future[Option[Preference]] =
     for {
       emailAddressCopied <- preferencesReceived.map(
                              _.map(pref => pref.copy(emailAddress = pref.email.map(_.email.value)))
                            )
-      statusPresent <- preferencesReceived.map(_.exists(_.email.isDefined))
+      statusPresent <- preferencesReceived.map(_.exists(_.status.isDefined))
       statusCopied <- if (statusPresent)
                        Future successful emailAddressCopied
-                         .map(pref =>
-                           pref.copy(status = Some(
-                             PaperlessStatus(name = mapStatusToExistingValue(pref.email.map(email => email.status)))
-                           )
-                           )
-                         )
+                         .map(pref => pref.copy(email = pref.email.map(_.copy(status = pref.status.get.name.get))))
```

Comment:
.get!!!

There should be no reason for these unless they have been guarded against or the field should not be optional.